src/mono/mono/mini/interp/transform.c

```diff
@@ -4815,6 +4816,8 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 	gboolean inlining = td->method != method;
 	InterpBasicBlock *exit_bb = NULL;
 
+	mono_interp_pgo_generate_start ();
```

Comment:
Why aren't these methods in `generate` directly ?

Reply:
They were originally browser-only until I realized I could make them all portable. If you want I can move those into transform.c

Reply:
I meant that `generate_code` is called recursively for every method that is inlined in the current method compilation. And it seems to me that you want to call this method only once per method compilation, in `generate` method.

Reply:
I misunderstood, great point. Would that mean I could get rid of the recursion handling too?

Reply:
yes